src/redux/ui/screens/data/handle-cta.ts

```diff
@@ -835,6 +849,125 @@ const handleCtaAction = async (
                     break;
                 }
 
+                case 'govProposalChoice': {
+                    const proposal = action.params?.params?.proposal;
+                    const choice = action.params?.params?.choice;
+                    const gzilContractAddress = action.params?.params?.gzilContractAddress;
+                    const metadata = action.params?.params?.metadata || {};
+                    const proposalType = action.params?.params?.proposalType || 'vote';
+
+                    const account = getSelectedAccount(state);
+                    const selectedWallet = getSelectedWallet(state);
+                    const appWallet = getWalletByPubKey(state, selectedWallet.walletPublicKey);
+
+                    if (!appWallet || !account) {
+                        break;
+                    }
+
+                    let password = '';
+
+                    if (appWallet.type === WalletType.HD) {
+                        password = await PasswordModal.getPassword(
+                            translate('Password.pinTitleUnlock'),
+                            translate('Password.subtitleSignMessage'),
+                            { sensitive: true, showCloseButton: true }
+                        );
+                        await LoadingModal.open({
+                            type: TransactionMessageType.INFO,
+                            text: TransactionMessageText.SIGNING
+                        });
+                    }
+
+                    const wallet: {
+                        signMessage: (
+                            blockchain: Blockchain,
+                            accountIndex: number,
+                            accountType: AccountType,
+                            message: string
+                        ) => Promise<any>;
+                    } =
+                        appWallet.type === WalletType.HW
```

Comment:
we need to test if message signing is working on ledger, this was not testet on SDK implementation